spec-tests/src/test/java/org/candlepin/spec/consumers/ConsumerResourceActivationKeySpecTest.java

```diff
@@ -103,144 +101,95 @@ public void shouldAllowConsumerToRegisterWithActivationKeys() {
+        int threadCount = 10;
+
+        OwnerDTO owner = this.ownerClient.createOwner(Owners.random());
+
+        // Create some activation keys and pools
+        ActivationKeyDTO[] keys = new ActivationKeyDTO[keyCount];
+        PoolDTO[] pools = new PoolDTO[poolCount];
+        for (int i = 0; i < keyCount; ++i) {
+            keys[i] = this.ownerClient.createActivationKey(owner.getKey(), ActivationKeys.random(owner));
+        }
 
-        int consumerCount = 3;
-        int poolCount = 6;
-        List<ConsumerDTO> consumers = Collections.synchronizedList(new ArrayList<>());
+        for (int i = 0; i < poolCount; ++i) {
+            ProductDTO product = this.ownerProductApi.createProduct(owner.getKey(), Products.random());
+            pools[i] = this.ownerClient.createPool(owner.getKey(), Pools.random(product));
+        }
 
-        // create 6 pools and 2 activation keys
-        ActivationKeyDTO key1 = ownerClient.createActivationKey(owner.getKey(),
-            ActivationKeys.random(owner));
-        ActivationKeyDTO key2 = ownerClient.createActivationKey(owner.getKey(),
-            ActivationKeys.random(owner));
+        // Attach the pools to the keys evenly, but random-ish-ly
+        Random rnd = new Random();
 
-        createProductsAndPoolsForActivationKeys(owner, key1, key2, poolCount);
-        List<PoolDTO> pools = ownerClient.listOwnerPools(owner.getKey());
-        assertThat(pools).hasSize(poolCount);
-
-        // register new systems
-        List<Thread> threads = new ArrayList<>();
-        IntStream.range(0, consumerCount).forEach(entry -> {
-            threads.add(new Thread(() -> {
-                String uuid = StringUtil.random("uuid");
-                ConsumerDTO consumer = Consumers.random(owner)
-                    .facts(Map.of("system.certificate_version", "3.3", "dmi.system.uuid", uuid));
-                consumer = consumerClient.createConsumer(consumer, null, owner.getKey(),
-                    key2.getName() + "," + key1.getName(), true);
-                // this ensures completion of creation before proceeding
-                consumer = consumerClient.getConsumer(consumer.getUuid());
-                assertThat(consumer).isNotNull();
-                assertThat(consumer.getEntitlementCount()).isEqualTo(poolCount);
-                consumers.add(consumer);
-            }));
-        });
-        threads.forEach(assertDoesNotThrow(() -> Thread::start));
-        for (Thread thread : threads) {
-            while (thread.isAlive()) {
-                thread.join();
-            }
+        // We can use a set seed if this test becomes inconsistent in the future, but so far the order just
+        // needs to be non-sequential and every iteration attempted has triggered the deadlock as expected
+        // long seed = -1753978213090389925L;
+        long seed = rnd.nextLong();
+        rnd.setSeed(seed);
+
+        System.out.printf("Shuffling pools using seed %d\n", seed);
```

Comment:
This might have been left from debugging

Reply:
Kind of, but I left it intentionally in the event we wanted to grab the seed for a given run. I suppose we can always put it back when need be.

Reply:
Okay! I'm fine with leaving it in